ethereum/contracts/ERC20App.sol

```diff
@@ -98,10 +97,7 @@ contract ERC20App is AccessControl {
         );
 
         balances[_token] = balances[_token] - _amount;
-        require(
-            IERC20(_token).transfer(_recipient, _amount),
-            "ERC20 token transfer failed"
-        );
+        IERC20(_token).safeTransfer(_recipient, _amount);
```

Comment:
This makes sense in the unlock function, but I'm also still not sure about if we should implement this or not. Plz read through and make sure you understand the following: https://samczsun.com/the-dangers-of-surprising-code/

Basically, safeTransfer introduces the potential for a re-entrancy attack. Having said that though, it is still nice to have permanent lockup protection. In our case I don't think there is any risk of a re-entrancy attack as this function is permissioned and uses check-effects-interaction, and so I think it's fine to keep safeTransfer.